feat!: migration from @effect/schema to effect monorepo package

totto2727 opened the pull request:
I have created a PR for the following Issue as I have not seen any progress on it.
If it has already been addressed, I will close it.

close https://github.com/ciscoheat/sveltekit-superforms/issues/497

# Changes

- Delete @effect/schema
- Upgrade @effect package
- Correction of import path

Local testing was successful.

ciscoheat commented:
Thank you, it's a good start, but there are typing problems with the adapter that I couldn't figure out easily. Maybe you know more about the new typing and can do it?

totto2727 commented:
The test was inadequate...
Was not working on dev server...

The following fixes have been added

- Import fixes
  - Added Either namespace to src/lib/adapters/effect.ts, removed Effect namespace
- Fixed some type references.
  - src/lib/adapters/typeSchema.ts

Dev server and type checking confirmed that there are no problems.

ciscoheat commented:
Great, I'll check it out locally soon, and if everything's fine I'll add it to the next release.
